packages/backend/src/graphql/queries/tiles/get-table-connections.ts

```diff
@@ -0,0 +1,53 @@
+import Flow from '@/models/flow'
+
+import type { QueryResolvers } from '../../__generated__/types.generated'
+
+import getTables from './get-tables'
+
+interface ExtendedFlow extends Flow {
+  tableid: string
+  count: number
+}
+
+interface TableConnection {
+  [key: string]: number
+}
+
+const getTableConnections: QueryResolvers['getTableConnections'] = async (
+  _parent,
+  { limit, offset, name },
+  context,
+) => {
+  // get tables of currentUser in the current page
+  const tables = await getTables(_parent, { limit, offset, name }, context)
+
+  if (tables.edges.length === 0) {
+    return {}
+  }
+  const tableIds = tables.edges.map((t) => t.node.id)
+  const tableIdStr = tableIds.map((id) => `'${id}'`).join(', ')
+
+  // get distinct rows of tables used in flows
+  // returns flow id and table id
```

Comment:
add a comment to monitor if query is too long, if so, add index on step.parameters ->> tableId